File: 48rotateImage/rotateImage.cpp
class Solution {
public:
    void rotate(vector<vector<int>>& matrix) {
        std::reverse(std::begin(matrix), std::end(matrix));
        for (std::size_t row{}; row != std::size(matrix); ++row)
            for (std::size_t column{row + 1}; column != std::size(matrix); ++column) std::swap(matrix.at(row).at(column), matrix.at(column).at(row));
    }
};

File: 48rotateImage/rotateImage.java
class Solution {
    public void rotate(int[][] matrix) {
        IntStream.range(0, matrix.length / 2).forEach($ -> {
            int[] tmp = matrix[$];
            matrix[$] = matrix[matrix.length - 1 - $];
            matrix[matrix.length - 1 - $] = tmp;});
        IntStream.range(0, matrix.length).forEach(row -> IntStream.range(row + 1, matrix.length).forEach(column -> {
            matrix[row][column] ^= matrix[column][row];
            matrix[column][row] ^= matrix[row][column];
            matrix[row][column] ^= matrix[column][row];}));
    }
}
